src/App.js

```diff
@@ -1,25 +1,149 @@
+            count: userSelectedCart[key] ? userSelectedCart[key].count : 0,
+
+            value:inventory[key].value
+            
+          }
+          )
+      }
+
+      for( let key in userSelectedCart){
+
+        newUserCart.push(
+          {
+            key,
+            title: userSelectedCart[key].title,
+            imgUrl:userSelectedCart[key].imgUrl,
+            price:userSelectedCart[key].price,
+            count: userSelectedCart[key].count,
+            value:userSelectedCart[key].value
+          })
+      }
+      setPictures(newInventory);
+      setUserCart(newUserCart);
+      
+    })
+  },[])
+
+  // event listener pass to header component, filter items based on value "coffee" or "beer"
+  const handleUserChoice = (e)=>{
+    const ItemsFilter = pictures.filter((item)=>{
+      return item.value === e.target.value
+    })
+    setFilteredItems(ItemsFilter)
+    
+  }
+  
+
+  // function when user click add to cart button
+  const add = function(userSelectItem){
+
+    const database = getDatabase(firebase);
+    const dbRef = ref(database, '/userSelectedCart');
+
+    const userSelectedItemInfo = {
+      [userSelectItem.key]:{
+        price: userSelectItem.price,
+        imgUrl: userSelectItem.imgUrl,
+        title: userSelectItem.title,
+        count: userSelectItem.count + 1
+      }      
+    }  
+    update(dbRef, userSelectedItemInfo)
+  }
+
+  // when user click delete from cart inside shopping cart Nav
+  const minus = function (userRemoveItem){
+    
+    const database = getDatabase(firebase);
+
+    const dbRef = ref(database, `userSelectedCart/${userRemoveItem}`);
+    remove(dbRef,userRemoveItem)
+  }
+
+  // function to change the state variable userCart from false to true 
+  const handleCart = function(){
+    setCart(!openCart)
+  }
+
+  // updated the shopping cart number 
+  const getTotalCount = function(shoppingCart){
+    let totalCount = 0;
+    shoppingCart.forEach((item)=>{
+      totalCount = totalCount + item.count
+      // for every item, increase the totalcount by the item.count
+    })
+    return totalCount;
+  }
+
+  // function to calculate total amount of items in shopping cart
+  const totalAmountCalculator = (userPickedItems)=>{
+    let totalAmount = 0
+    userPickedItems.forEach((item)=>{
+      totalAmount = totalAmount + item.price * item.count
+    })
+    return totalAmount
+  }
+
+
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+
+      <Header handleCart={handleCart} totalCount ={getTotalCount(userCart)} pictures={pictures} handleUserChoice={handleUserChoice}/>
```

Comment:
Love the design and functionality of the cart! Perhaps add a circular background behind the shopping cart or something as the cart blends in with the darker product images when scrolling through them.